### DISCUSSION TITLE
New Django app using ThumbmarkJS

### DISCUSSION THREAD

**Discussion by markciecior** [thumbmarkjs/thumbmarkjs#78] [Show and tell]
Hi All,

I've been looking for a way to uniquely identify users for a short-lived, public, anonymous survey I run.  The survey was built using Django and some users have figured out that clearing their cookies and/or using Private Mode allows them to submit multiple responses.

I came across this library which seems to be exactly what I need to identify users and prevent them from submitting more than once.  I created a reusable Django app that implements ThumbMarkJS and ties anonymous HTTP requests into Django's authentication system.  Now with a single decorator I can turn anonymous users into "logged-in" users and prevent them from submitting again.

I realize there are a few shortcomings and the identification isn't 100% unique, but this library has almost completely solved my problem.  The app is available on PyPi [here](https://test.pypi.org/project/django-thumbmark/) if anyone's interested.

**Comment by ilkkapeltola** [comment]
This is really cool! 👍
